worker/src/main/scala/org/apache/celeborn/service/deploy/worker/storage/StorageManager.scala

```diff
@@ -358,13 +358,14 @@ final private[worker] class StorageManager(conf: CelebornConf, workerSource: Abs
       if (dirs.isEmpty) {
         val shuffleDir =
           new Path(new Path(hdfsDir, conf.workerWorkingDir), s"$appId/$shuffleId")
-        FileSystem.mkdirs(StorageManager.hadoopFs, shuffleDir, hdfsPermission)
         val fileInfo =
           new FileInfo(
             new Path(shuffleDir, fileName).toString,
             userIdentifier,
             partitionType,
             partitionSplitEnabled)
+        fileInfos.computeIfAbsent(shuffleKey, newMapFunc).put(fileName, fileInfo)
```

Comment:
Before this PR, `fileInfos` will not put a new object if Exception is thrown. So I think we'd better try to delete the `fileName` from `fileInfos` where `createWriter` is called, i.e. in the catch block of `Controller.handleReserveSlots`

Reply:
During the heartbeat between master and worker, the active ShuffleKeys of the worker are compared with the registered ShuffleKeys of the master, and expired ones are added to the expired ShuffleKeys for automatic cleaning.